nautobot/extras/models/customfields.py

```diff
@@ -724,7 +724,20 @@ def delete(self, *args, **kwargs):
 
         super().delete(*args, **kwargs)
 
-        delete_custom_field_data.delay(self.key, content_types)
+        # Circular Import
+        from nautobot.extras.signals import _get_user_if_authenticated, change_context_state
+
+        change_context = change_context_state.get()
+        if change_context is None:
+            context = {}
+        else:
+            context = {
+                "user": _get_user_if_authenticated(change_context.get_user(), self),
+                "change_id": change_context.change_id,
+                "context_detail": "delete custom field data",
+                "context": change_context.context,
+            }
```

Comment:
How about adding an `.as_dict()` method to `ChangeContext`? (I don't know that we need different context_detail strings for each of these methods, it *might* be preferable to keep whatever context_detail was already set on the ChangeContext itself instead?)

Reply:
The context detail is "" on these ChangeContext. It might be helpful to be more specific?